worlds/yoshisisland/Regions.py

```diff
@@ -0,0 +1,211 @@
+        create_region(multiworld, player, locations_per_region, location_cache, "4-2"),
+        create_region(multiworld, player, locations_per_region, location_cache, "4-3"),
+        create_region(multiworld, player, locations_per_region, location_cache, "4-4"),
+        create_region(multiworld, player, locations_per_region, location_cache, "Marching Milde's Boss Room"),
+        create_region(multiworld, player, locations_per_region, location_cache, "4-5"),
+        create_region(multiworld, player, locations_per_region, location_cache, "4-6"),
+        create_region(multiworld, player, locations_per_region, location_cache, "4-7"),
+        create_region(multiworld, player, locations_per_region, location_cache, "4-8"),
+        create_region(multiworld, player, locations_per_region, location_cache, "Hookbill The Koopa's Boss Room"),
+
+        create_region(multiworld, player, locations_per_region, location_cache, "5-1"),
+        create_region(multiworld, player, locations_per_region, location_cache, "5-2"),
+        create_region(multiworld, player, locations_per_region, location_cache, "5-3"),
+        create_region(multiworld, player, locations_per_region, location_cache, "5-4"),
+        create_region(multiworld, player, locations_per_region, location_cache, "Sluggy The Unshaven's Boss Room"),
+        create_region(multiworld, player, locations_per_region, location_cache, "5-5"),
+        create_region(multiworld, player, locations_per_region, location_cache, "5-6"),
+        create_region(multiworld, player, locations_per_region, location_cache, "5-7"),
+        create_region(multiworld, player, locations_per_region, location_cache, "5-8"),
+        create_region(multiworld, player, locations_per_region, location_cache, "Raphael The Raven's Boss Room"),
+
+        create_region(multiworld, player, locations_per_region, location_cache, "6-1"),
+        create_region(multiworld, player, locations_per_region, location_cache, "6-2"),
+        create_region(multiworld, player, locations_per_region, location_cache, "6-3"),
+        create_region(multiworld, player, locations_per_region, location_cache, "6-4"),
+        create_region(multiworld, player, locations_per_region, location_cache, "Tap-Tap The Red Nose's Boss Room"),
+        create_region(multiworld, player, locations_per_region, location_cache, "6-5"),
+        create_region(multiworld, player, locations_per_region, location_cache, "6-6"),
+        create_region(multiworld, player, locations_per_region, location_cache, "6-7"),
+        create_region(multiworld, player, locations_per_region, location_cache, "6-8"),
+        create_region(multiworld, player, locations_per_region, location_cache, "Bowser's Room"),
+
+    ]
+    if get_option_value(multiworld, player, "extras_enabled") == 1:
+        regions.insert(68, create_region(multiworld, player, locations_per_region, location_cache, "6-Extra")),

```

Comment:
They all have to be like that so they can show up correctly in the level order. :/

Reply:
Fair enough. I'm trying to think if there is a better way to do this because this is highly vulnerable to bugs if you ever change the region list for whatever reason

Reply:
Hold up one second. Do these regions have to be conditionally created anyway?

You're both: 1. Conditionally adding the bonus locations to location_table
But also: 2. Adding the bonus regions containing those locations conditionally to multiworld.regions

If you removed these conditions and just put them in the list, it'd create the regions, but they'd be empty. Which wouldn't be too bad. I guess it might have a slight performance impact?